### ISSUE TITLE
VS Code crashing on startup

### ISSUE THREAD

**Issue by julesx** [daishi4u/ForceCode#7]
I start VS Code, and I see some toasts 

![image](https://user-images.githubusercontent.com/3010047/42517005-c50ce530-842c-11e8-8eff-1286497dd6ca.png)

i see the max of 3 toasts, and then my VS Code locks up.  I have not changed these files.  My only option is to force close VS.

Sometimes I am able to dismiss a few toasts before it locks up.  More just appear.

**Comment by daishi4u** [comment]
What do they say? It may be an issue with vscode...

**Comment by julesx** [comment]
Can you not see the embedded image? They are all notifications that my .cls files have changed. They have not.

**Comment by daishi4u** [comment]
Sorry the images didn't show up in the app I was using. Try deleting wsMembers.json in the project folder and starting vscode again. Let me know if this works. Sometimes the dates on the salesforce files are a bit off so you'll get a notification of changed files even though you haven't changed it since last load. Right now I allow a +/-2 seconds, but maybe that's not enough. Either way, this shouldn't crash vscode, I've tested it on windows 10, 7, and Ubuntu

**Comment by julesx** [comment]
I think the problem is that it is attempting to notify for every single file, and there are a fair number of them.

**Comment by daishi4u** [comment]
No it's not needed for code coverage. That file is only used to detect changes to files. I have pushed an update (3.3.2) that ups the time difference for checking for changed files and also adds an option to turn off checking for changed files on load. If you want to still check for changes on load then you will need to add the option "checkForFileChanges": true to your force.json file. If you leave it out of the config or set it to false then it will skip the check.

**Comment by daishi4u** [comment]
Version 3.3.3 adds checking by comparing user Ids as well as an extra precaution. If you decide to upgrade then you will need to delete wsMembers.json before starting vsCode after installing the extension, otherwise it will tell you every single file has been changed. Let me know if you face any other issues with this after upgrading to either 3.3.2 or 3.3.3.

**Comment by julesx** [comment]
what is the purpose of this file change check?

**Comment by daishi4u** [comment]
To inform you of files that have been changed by other people so that your source files don't become stale. Say that someone changes something you've been working on, now you come back to this file to make changes without knowing this other person did, if this check wasn't in place you could end up overwriting their work with old data and completely wiping it out. Make sense?

**Comment by daishi4u** [comment]
I will be limiting the maximum number of running tasks to ten in my next release, as this will help with this type of issue as well. I have a way to show all tasks, so say by chance there's 30 of them then I will show them all as running, but only execute 10 at a time until they all complete. Probably more info than you care to know, but this is also a reminder to how I want to go about it.

**Comment by julesx** [comment]
So it checks on the server side if a file has changed since you last downloaded it?

Does this happen only at startup?  It would probably be most useful on save.

**Comment by daishi4u** [comment]
Yes it checks on startup so that you can refresh any stale files and also checks on save.
